beacon-chain/rpc/validator/proposer.go

```diff
@@ -356,11 +356,18 @@ func (vs *Server) canonicalEth1Data(ctx context.Context, beaconState *stateTrie.
 		canonicalEth1Data = beaconState.Eth1Data()
 		eth1BlockHash = bytesutil.ToBytes32(beaconState.Eth1Data().BlockHash)
 	}
-	_, latestEth1DataHeight, err := vs.Eth1BlockFetcher.BlockExists(ctx, eth1BlockHash)
+	_, latestEth1Height, err := vs.Eth1BlockFetcher.BlockExists(ctx, eth1BlockHash)
 	if err != nil {
 		return nil, nil, errors.Wrap(err, "could not fetch eth1data height")
 	}
-	return canonicalEth1Data, latestEth1DataHeight, nil
+
+	canonicalEth1Height := big.NewInt(0)
```

Comment:
this whole chunk of logic is incorrect, this will end up reverting eth1data back 1024 blocks.